### PR TITLE
Debug mode for Inbox tab GBR doesn't handle multi-policy

### PR DESCRIPTION
**Pull request by pac-guerreiro**
### Fixed Issues

$https://github.com/Expensify/App/issues/51271
PROPOSAL:
- Add reportID to optimistic reports in order to fix debug mode showing reports with undefined reportID in url
- Also map existing reports with no reportID with their collection key
- Debug mode now redirects user to debug report page instead of report page to help diagnose what's happening and allow for deletion
- Debug report page now shows a `View Report` button to redirect the user to the report page. This is useful when we don't access the debug report page from the report page.

### Tests

On **web**
1. Open a report that doesn't exist from the url (e.g. `r/123456`)
2. The view above the bottom navigation bar will show up a message saying `There's a report with errors`
3. Clicking on `View` button will open the RHP with the debug report page with the data of the offending report
4. Deleting the report will clear the RBR

### Offline tests

Same as tests

### QA Steps

Same as tests

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/e042d7a4-56c0-4b15-8963-e671fd1069cb

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/cd9284b5-07df-4275-b34e-cf13605b6041

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/5f2e8cce-80f9-4f12-a6ba-50ed0d085e28

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

### CODE REVIEW

**Comment by pac-guerreiro** [comment]
Today's update:
- [x] Add reportID to optimistic report
- [x] Extract reportID from report key to prevent redirecting the user to reports with undefined IDs
- [x] Add `View Report` button in debug report page

Todo:
- [ ] Prevent report from being re-fetched after deleting it from debug report page
- [ ] Add screen recordings
- [ ] Add QA steps

**Comment by pac-guerreiro** [comment]
@puneetlath @DylanDylann 

I'm setting this PR as ready for review although it's missing QA steps and screen recordings for native app.

I'll add these next Monday!

**Comment by pac-guerreiro** [comment]
@puneetlath @DylanDylann 

I wasn't able to find steps to test this on native platforms yet. Will continue my research tomorrow!

**Comment by DylanDylann** [comment]
We are displaying the error indicator on LHN even though there are no error reports. It confused me

Screenshot 2024-10-29 at 16 12 59

**Comment by DylanDylann** [comment]
On LHN, we use useReportIDs hook to filter and sort reports first

https://github.com/Expensify/App/blob/2463bf74b4edabdde887636f7874fdbbfcf9eb2c/src/pages/home/sidebar/SidebarLinksData.tsx#L40

I think we should apply the logic in useReportIDs into getChatTabBrickRoadReport to make it consistent

**Comment by DylanDylann** [comment]
> I wasn't able to find steps to test this on native platforms yet. Will continue my research tomorrow!

- Android: open the deep link from Terminal: npx uri-scheme open new-expensify://concierge --android
- iOS: open the deep link from Terminal: npx uri-scheme open new-expensify://concierge --ios
- Desktop: open the deep link from console: window.location.href="/settings/workspaces"

Anyways, I can help you with this step

**Comment by pac-guerreiro** [comment]
> We are displaying the error indicator on LHN even though there are no error reports. It confused me

This is known and will be addressed here - https://github.com/Expensify/App/issues/51488

**Comment by pac-guerreiro** [comment]
> > I wasn't able to find steps to test this on native platforms yet. Will continue my research tomorrow!
> 
> * Android: open the deep link from Terminal: npx uri-scheme open new-expensify://concierge --android
> * iOS: open the deep link from Terminal: npx uri-scheme open new-expensify://concierge --ios
> * Desktop: open the deep link from console: window.location.href="/settings/workspaces"
> 
> Anyways, I can help you with this step

Oh nice, today I learned 😄

But I was aiming for reproduction steps inside the app it self as an user, do you know how to use deep links without the terminal or console?

Also there's a case that I am not able to reproduce which is a workspace chat that existed but then the user started receiving not found errors.

Here is the onyx data for a report where this happens:

```
"report_8056514671453445": {
    "chatReportID": "5966879180236894",
    "chatType": "",
    "currency": "USD",
    "description": "",
    "errorFields": {
      "notFound": {
        "1729100863505463": "Report not found",
        "1729614594061340": "Report not found",
        "1729614643262146": "Report not found",
        "1729615320612665": "Report not found",
        "1729615341326930": "Report not found",
        "1729615344962131": "Report not found",
        "1729616332028548": "Report not found",
        "1729616365333863": "Report not found"
      }
    },
    "hasOutstandingChildRequest": false,
    "hasOutstandingChildTask": false,
    "hasParentAccess": true,
    "isCancelledIOU": false,
    "isOwnPolicyExpenseChat": false,
    "isPinned": false,
    "isWaitingOnBankAccount": false,
    "lastActionType": "IOU",
    "lastActorAccountID": "12032073",
    "lastMessageHtml": "$0.00 expense",
    "lastMessageText": "$0.00 expense",
    "lastReadSequenceNumber": 0,
    "lastVisibleActionCreated": "2024-10-16 17:42:06.439",
    "lastVisibleActionLastModified": "2024-10-16 17:42:06.439",
    "nonReimbursableTotal": 0,
    "notificationPreference": "hidden",
    "oldPolicyName": "",
    "parentReportActionID": "3470603815071771046",
    "parentReportID": "5966879180236894",
    "permissions": [],
    "private_isArchived": "",
    "reportID": "8056514671453445",
    "reportName": "Expense Report #8056514671453445",
    "stateNum": 1,
    "statusNum": 1,
    "total": -13794,
    "type": "expense",
    "unheldTotal": -13794,
    "welcomeMessage": "",
    "writeCapability": "all"
  }
```

File: src/pages/Debug/DebugDetails.tsx [hunk 1]
```diff
@@ -236,7 +237,10 @@ function DebugDetails({data, children, onSave, onDelete, validate}: DebugDetails
                         text={translate('common.delete')}
                         onPress={() => {
                             onDelete();
-                            Navigation.goBack();
+                            // Dismiss the current report screen and replace it with Concierge Chat
```
Comment (DylanDylann) [inline src/pages/Debug/DebugDetails.tsx:240]: @pac-guerreiro Why should we do that?

File: src/libs/actions/Report.ts [hunk 2]
```diff
@@ -767,10 +767,13 @@ function openReport(
         return;
     }
 
-    const optimisticReport = reportActionsExist(reportID)
-        ? {}
+    const optimisticReport: Report = reportActionsExist(reportID)
+        ? {
+              reportID,
+          }
```
Comment (DylanDylann) [inline src/libs/actions/Report.ts:773]: this change seems not safe to me. I think this change is unnecessary if we apply [this approach](https://github.com/Expensify/App/issues/51488#issuecomment-2443758461). Anyways, could we handle it in https://github.com/Expensify/App/issues/51488 and revert this change? Thanks

File: src/libs/WorkspacesSettingsUtils.ts [hunk 3]
```diff
@@ -125,8 +125,14 @@ function getChatTabBrickRoadReport(policyID?: string): OnyxEntry<Report> {
         return undefined;
     }
 
+    // There are optimistic reports without a reportID, so we need to extract it from the report key
+    const normalizedAllReports = Object.entries(allReports).map(([key, report]) => ({
+        ...report,
+        reportID: report?.reportID ?? key.replace(ONYXKEYS.COLLECTION.REPORT, ''),
+    }));
+
```
Comment (DylanDylann) [inline src/libs/WorkspacesSettingsUtils.ts:133]: Same. I think this change is unnecessary if we apply https://github.com/Expensify/App/issues/51488#issuecomment-2443758461. Anyways, could we handle it in https://github.com/Expensify/App/issues/51488 and revert this change? Thanks

File: src/pages/Debug/DebugDetails.tsx (hunk 1 above)
Comment (pac-guerreiro) [reply src/pages/Debug/DebugDetails.tsx:240]: @DylanDylann This is a workaround I found to prevent the app from fetching the report data again, it's based on this:

Screenshot 2024-10-29 at 10 21 22

But now that I'm seeing this again, I think I should just use `Report.navigateToConciergeChatAndDeleteReport` instead of duplicating code

**Comment by pac-guerreiro** [comment]
@DylanDylann I just reverted the commits that added reportID to reports that were missing it

**Comment by pac-guerreiro** [comment]
@DylanDylann I replaced the old logic to navigate after deleting a report with `navigateToConciergeChatAndDeleteReport` function

**Comment by DylanDylann** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Review by DylanDylann — APPROVED** [review]

**Review by puneetlath — APPROVED** [review]
